### ISSUE TITLE
Modifiers in the choice annotation are ignored when a model is redeclared by using the parameters pop-up window (it does not where an instance of the model is redeclared)

### ISSUE THREAD

**Issue by AndreaBartolini** [OpenModelica/OpenModelica#12087]
### Description
If some modifiers are written in the choice annotation of a replaceable model they are ignored when the model is redeclared by using the parameters pop-up window, if the replaceable element is not a model but is an instance of the model (component) then they are considered.

### Steps to Reproduce
Consider the MWE in the following (also attached)

```modelica
package TestChoice
  model BaseM
    parameter Real pBase = 0;
  end BaseM;
  
  model ExtM
    extends BaseM;
    parameter Real pExt = 0;
  end ExtM;

  model RepM
    replaceable BaseM rM1
    annotation(choices(choice(redeclare ExtM rM1(pBase=7, pExt=5))));
  
    replaceable model RM = BaseM
    annotation(choices(choice(redeclare model RM = ExtM(pBase=7, pExt=5))));
    RM rM2;
  end RepM;
  
  model Test
    RepM repM  annotation(
      Placement(transformation(extent = {{-10, -10}, {10, 10}})));
  end Test;
end TestChoice;
```

1) open the MWE in OMEdit
2) open and run the `Test `model, the result (as expected) is that both `rM1 `and `rM2 ` have the parameter `pBase=0 `
3) open the parameters pop-up of the instance `repM `and redeclare both the `rM1 `and `rM2 `by selecting the available choices, the choice of the instance `rM1 `contains the modifiers
![immagine](https://github.com/OpenModelica/OpenModelica/assets/25974985/8dcac894-69e6-4cf4-b7e1-ef2bd64d35dd)

but the same of the model `RM `does not
![immagine](https://github.com/OpenModelica/OpenModelica/assets/25974985/8856005e-bf37-4cf8-bcbd-b0159f5e27ab)

here the result in the text view after the redeclarations:
```modelica
model Test
  RepM repM(redeclare ExtM rM1(pBase = 7, pExt = 5), redeclare model RM = ExtM)   annotation(
    Placement(transformation(extent = {{-10, -10}, {10, 10}})));
end Test;
```
and, obviously, If you run the model both the `rM1 `and `rM2 `have the right parameters (`pBase `and `pExt`) but only the ones of `rM1` have the right values (7 and 5 respectively).

### Expected Behavior
the model Test after the redeclaration appears as follows (by excluding the text formatting):
```modelica
model Test
  RepM repM(redeclare ExtM rM1(pBase = 7, pExt = 5), 
            redeclare model RM = ExtM(pBase = 7, pExt = 5))   
       annotation(
    Placement(transformation(extent = {{-10, -10}, {10, 10}})));
end Test;
```

### Version and OS
OpenModelica v1.23.0-dev-354-g23752f8f3b (64-bit)
Win11 Pro (64bit)

[TestChoice.zip](https://github.com/OpenModelica/OpenModelica/files/14523263/TestChoice.zip)

**Comment by AndreaBartolini** [comment]
In addition,
consider the expected result:

```modelica
model Test
  RepM repM(redeclare ExtM rM1(pBase = 7, pExt = 5), 
            redeclare model RM = ExtM(pBase = 7, pExt = 5))   
       annotation(
    Placement(transformation(extent = {{-10, -10}, {10, 10}})));
end Test;
```

if you open the parameters pop-up of the `repM `instance the modifiers in the textual code are not shown for class RM, nor in the redeclaration row neither in the dedicated sub-window (see below)

![immagine](https://github.com/OpenModelica/OpenModelica/assets/25974985/549b51d1-d906-4231-a3a7-2fbc586ad9c5)

I suppose this is related to the same issue, is just to add an information and a check for the bug-fixing.

**Comment by AndreaBartolini** [comment]
keep @cearolo in copy

**Comment by adeas31** [comment]
Fixed with #12094 and ported to maintenance 1.22 #12095
